### PR TITLE
:memo: Add comprehensive testing instructions for standardized test development

### PR DESCRIPTION
**Pull request by Mistifiou**
## 🧪 Testing Instructions Implementation

This PR introduces comprehensive testing guidelines to establish standardized test development across the project.

### 📋 What's Added

- **Testing instruction file**: `.github/instructions/tester.instructions.md`
- **Scope**: Applied to all `src/**/**.test.tsx` files
- **Length**: Concise 113-line format for maintainability

#### Core Standards
- **English-only** descriptions and comments
- **PascalCase** naming: `FeatureName.test.tsx`
- **Organized folder structure**: `rendering/`, `validation/`, `submission/`, `interaction/`, `fixtures/`

#### Standard Patterns
- **i18next mocking** - Unified mock implementation across all test files
- **Faker data generation** - Realistic test data creation with @faker-js/faker
- **Zod schema testing** - Validation testing patterns for form schemas
- **Form utilities** - Reusable test helpers for form interaction

#### Test Categories
- **Validation Tests**: One test per rule, edge cases, error display/removal timing
- **Form Submission**: Success/error/loading scenarios with proper state management
- **Rendering Tests**: Element presence, input types, visual indicators
- **Interaction Tests**: Focus management, keyboard navigation, blur validation

#### Quality Criteria
- **100% validation branch coverage** requirement
- **<100ms performance** benchmark per individual test
- **No flaky tests** with proper mock cleanup using `mockRestore()`
- **Maintainable code** via centralized fixtures and utilities

#### Forbidden Patterns
- ❌ Copy-paste code between test files
- ❌ Hardcoded test values (must use Faker generators)
- ❌ Test files exceeding 110 lines
- ❌ Different i18next mock implementations per file
- ❌ `any` types in test helpers and utilities

#### Required Test Coverage
**Per validation field:**
1. ✅ Positive validation (valid values accepted)
2. ✅ Negative validation (invalid values rejected)
3. ✅ Error clearing (error disappears when corrected)
4. ✅ Validation timing (only triggered after blur event)
5. ✅ Edge cases (min/max values, special formats)

**Per form component:**
1. ✅ Successful submission (valid data sent correctly)
2. ✅ Blocked submission (validation errors prevent sending)
3. ✅ Loading states (form disabled during submission)
4. ✅ Error handling (network/server errors managed gracefully)
5. ✅ Type transformation (string→number conversions, etc.)

### 💡 Impact & Benefits

- **Standardizes** test development patterns across the entire team
- **Improves** test maintainability and code readability
- **Reduces** code duplication through shared utilities and fixtures
- **Ensures** comprehensive validation coverage for all form components
- **Establishes** clear performance and quality benchmarks
- **Provides** consistent testing approach for Zod form validation

### 🔗 Integration

- Integrates seamlessly with existing Zod form validation infrastructure
- Complements current coding standards and development workflow
- Supports the comprehensive form component library architecture
- Aligns with BrutalComet coding philosophy and quality standards

### 📚 Usage

Apply these instructions when working on any file matching `src/**/**.test.tsx`. The guidelines ensure consistent, maintainable, and comprehensive test coverage across all form components and validation logic.

---

**Ready to transform our test suite from amateur hour to professional excellence!** 🔥

### CODE REVIEW

File: .github/instructions/tester.instructions.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+## 🔧 Utilities
+
+[WIP]
+
+## 🚨 Forbidden Patterns
+
+❌ **Never**:
+- Copy-paste code between tests
+- Hardcode test values (use Faker)
+- Put multiple `describe()` blocks in one file
```
[the remark this answers is not in the document]
Comment (Mistifiou) [reply .github/instructions/tester.instructions.md]: > ❌ **Never**:
> - Copy-paste code between tests
> - Hardcode test values (use Faker)
> - Put multiple `describe()` main blocks in one file

serait ok?
